src/PowerShell/Microsoft.WinGet.Client.Engine/PSObjects/PSSourceResult.cs

```diff
@@ -39,5 +39,15 @@ internal PSSourceResult(Management.Deployment.PackageCatalogReference catalogRef
         /// Gets the type of the source.
         /// </summary>
         public string Type { get; private set; }
+
+        /// <summary>
+        /// Gets the trust level of the source.
+        /// </summary>
+        public string TrustLevel { get; private set; }

```

Comment:
Why is this a `string` and not a `PSSourceTrustLevel`?

Maybe that is normal in PowerShell, it just seems odd to me.

Reply:
The main reason is not load any dependency when the CLR loads this binary as the ALC hasn't been being configured yet at this point. For cswinrt enums, there are PSSomething equivalents that get converted to string and then converted to the cswinrt representation in the engine module. Since this doesn't have it he had to go with strings,

Reply:
Is it that we can't have a dependency inversion with `Engine` taking a dependency on `Cmdlets`?  And we can't have this kind of immediate dependency on `Engine` in types exposed on the interface of any cmdlet because we don't have the ALC yet to resolve it (thus preventing us from putting `PSSourceTrustLevel` in `Engine`)?

In any case, I don't have a problem with it being a string.  Just wasn't clear why.

Reply:
Yeah, we don't want to have the dependency on Engine until the ALC is setup